doc/spec/manifest.md

```diff
@@ -0,0 +1,214 @@
+# Distribution Content Manifests
+
+A Content Manifest is a simple JSON file which contains general fields that
+are typical to any package management system. The goal is for these manifests
+to describe an application and its dependencies in a content-addressable and
+verifiable way.
+
+## Field Descriptions
+
+- **`schemaVersion`** *int*
+
+    Specifies the schema version of the manifest as an integer. This document 
+    describes version `2` only.
+
+- **`createdAt`** *string*
+
+    ISO-8601 formatted combined date and time at which the manifest was
+    created.
+
+- **`maintainers`** *array*
+
+    An array of objects representing persons or organizations which are
+    designated as mainainers of the described content. Fields are:
+    
+    - **`name`** *string*
+    
+        The name of a maintaining person or organization.
+    
+    - **`email`** *string*
+    
+        The email address of a maintainer.
+    
+    - **`url`** *string*
+    
+        A URL to a maintainer's website, blog, etc.
+
+- **`description`** *string*
+
+    A brief description of the content or application.
+
+- **`keywords`** *array*
+
+    An array of strings which should be considered keywords that may be used
+    by a separate search system to index content.
+
+- **`website`** *string*
+
+    A URL to a related website or project page for the application.
+
+- **`licenses`** *array*
+
+    An array of strings which identify any software licenses which apply
+    to the application. Values should be a [SPDX License Identifier](https://spdx.org/licenses/).
+
+- **`dependencies`** *array*
+
+    An array of objects which describe content which this application depends
+    on. Those may be generic blobs of data or other content manifests and
+    their dependencies as described by their media type. Fields are:
+    
+    - **`mediaType`** *string*
+    
+        The MIME type of the dependency object.
+    
+    - **`size`** *int*
+    
+        The size in bytes of the dependency object.
+    
+    - **`repository`** *string*
```

Comment:
Thanks for the pointer/info.  I am curious though… what’s the advantage of using a string/namespace instead of just a real URL? The spec/examples even talk about converting it to a URL so what’s the advantage of this level of indirection?

-Doug

> On Mar 2, 2015, at 12:55 PM, Josh Hawn <notifications@github.com> wrote:
> 
> In doc/spec/manifest.md <https://github.com/docker/distribution/pull/62#discussion_r25620394>:
> 
> > +
> > +- **`dependencies`** *array*
> > +
> > +    An array of objects which describe content which this application depends
> > +    on. Those may be generic blobs of data or other content manifests and
> > +    their dependencies as described by their media type. Fields are:
> > +    
> > +    - **`mediaType`** *string*
> > +    
> > +        The MIME type of the dependency object.
> > +    
> > +    - **`size`** *int*
> > +    
> > +        The size in bytes of the dependency object.
> > +    
> > +    - **`repository`** *string*
> @ncdc <https://github.com/ncdc> is correct. The repository value is not a URL. You can think of it instead like a package name. It's a namespace in which content-addressable objects are stored. The spec for exactly what it is and how it's used is here: #179 <https://github.com/docker/distribution/pull/179>
> —
> Reply to this email directly or view it on GitHub <https://github.com/docker/distribution/pull/62/files#r25620394>.
>